File: server/todolistJava01/src/main/java/jp/co/linnefromice/todolist01/service/TaskService.java
package jp.co.linnefromice.todolist01.service;

import jp.co.linnefromice.todolist01.model.Task;
import jp.co.linnefromice.todolist01.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class TaskService {

    @Autowired
    private TaskRepository repository;

    public Optional<Task> findOne(final String id) {
        return repository.findById(id);
    }

    public List<Task> findAll() {
        return repository.findAll();
    }

    public Task save(Task task) {
        return repository.save(task);
    }
}

File: server/todolistKotlin01/src/main/kotlin/jp/co/linnefromice/todolistKotlin01/service/TaskService.kt
package jp.co.linnefromice.todolistKotlin01.service

import jp.co.linnefromice.todolistKotlin01.model.Task
import jp.co.linnefromice.todolistKotlin01.repository.TaskRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class TaskService (
    private val taskRepository: TaskRepository){

    fun findOne(id: String): Optional<Task> {
        return taskRepository.findById(id);
    }

    fun findAll(): List<Task> {
        return taskRepository.findAll();
    }

    fun save(task: Task): Task {
        return taskRepository.save(task);
    }
}